### ISSUE TITLE
Glitched box drawing characters at tiny size

### ISSUE THREAD

**Issue by arcanis** [mintty/mintty#634]
I'm running the master branch (I wanted to try the #628 fix).

When the previous or next characters are "─", the "┌" and "┐" characters are glitched (they appear to be somewhat rounded instead of a real 90 degrees corner). Strangely, the opposite characters are displayed correctly:

![image](https://cloud.githubusercontent.com/assets/1037931/23102665/4bf9eaaa-f6a5-11e6-9187-d5e57879efb4.png)

I've double-checked that I'm using the right unicode characters, and everything seems correct. When increasing the zoom level via <kbd>Ctrl</kbd>+<kbd>+</kbd> the rendering is fine, but when resetting it with <kbd>Ctrl</kbd>+<kbd>0</kbd> it breaks again.

You can test it via:

```
$> printf ┌──────────┐
```

**Comment by mintty** [comment]
Which font/size?

**Comment by arcanis** [comment]
Consolas 9pt

**Comment by mintty** [comment]
Cannot reproduce. Please provide your mintty config file and the version number of your Consolas font.

**Comment by arcanis** [comment]
My .minttyrc:

```
BoldAsFont=-1
Font=Consolas
CursorType=block
BellType=0
BellFlash=no
Term=xterm-256color
FontSmoothing=default
```

My font is reported as being v6.96.

**Comment by mintty** [comment]
Oh, I see it when I change the zooming factor (~DPI scaling) to 100%. I don't see it with 9pt at 125% but it appears with 7pt at 125%.
I haven't seen it yet with any other font.
It does not happen either if I revert the auto-row-spacing specifically for Consolas with `-o RowSpacing=-2`.
Weird glitch indeed, don't know what to do about it.

**Comment by mintty** [comment]
This is actually a font problem, or a font handling problem induced by Windows.
Copy the test text into Windows notepad, configure the same font/size to verify.
Same effect; closing.

**Comment by adisib** [comment]
> Copy the test text into Windows notepad, configure the same font/size to verify.

This gives a different result between mintty and notepad for me. Notepad has no issues rendering the line.
![https://s31.postimg.org/6beeg9laj/fontissue.png](https://s31.postimg.org/6beeg9laj/fontissue.png)

**Comment by mintty** [comment]
This is a different test case. It looks like a frame displayed with VT100 graphics characters.
Those have another bug indeed which I've identified already.
To check this out, please test with the string behind "printf" above, and you may also make a log of the frame you displayed.

**Comment by adisib** [comment]
The printf test case above does the same. Broken in mintty, not in notepad.
![https://s31.postimg.org/inzocp7t7/fontissue2.png](https://s31.postimg.org/inzocp7t7/fontissue2.png)

**Comment by mintty** [comment]
This seems to be also affected by the screen scaling factor (DPI setting).
Try to reproduce at 100%.

**Comment by BrianInglis** [comment]
Reproducible with Consolas at smaller odd sizes at 100%. 
Start with Consolas 12pt, printf a few times,  ctrl-- to zoom smaller and you can see it at the equivalent zoom levels. 
May be hints of it also with Lucida Console at unreasonably small sizes, not other fonts, from what I see. 
May be a baseline hint or x-size problem with Consolas at small odd sizes?

**Comment by adisib** [comment]
DPI was always at 100% for me.

**Comment by mintty** [comment]
I've seen it in mintty with DejaVu Sans Mono 7pt and Consolas 9pt, 100% screen scaling, reproducible in notepad. With 125% screen scaling, notepad seems to deviate (also in visual font size) which is likely due to different DPI handling approaches.
It may well be an effect of hinting. Anyway, as I said above, it's nothing in the scope of mintty.

The VT100 graphics glitch in frame borders is a different thing and it's related to character repetition (the DEC REP sequence); I'll have a solution for *that one* soon.

**Comment by mintty** [comment]
Fixed the VT100 graphics and non-BMP glitches when DEC REP (repeat previous character) function is used.

**Comment by mintty** [comment]
Released 2.8.3.
